catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -222,20 +284,45 @@ const useBucketFieldsStyles = M.makeStyles((t) => ({
   panelSummaryContent: {
     margin: `${t.spacing(1)}px 0 !important`,
   },
+  warning: {
+    background: t.palette.warning.main,
+    marginBottom: t.spacing(1),
+    marginTop: t.spacing(2),
+  },
+  warningIcon: {
+    color: t.palette.warning.dark,
+  },
 }))
 
 interface BucketFieldsProps {
-  name?: string
+  bucket?: BucketConfig
   reindex?: () => void
 }
 
-function BucketFields({ name, reindex }: BucketFieldsProps) {
+function BucketFields({ bucket, reindex }: BucketFieldsProps) {
   const classes = useBucketFieldsStyles()
+
+  const [{ error, data }] = urql.useQuery({ query: CONTENT_INDEXING_SETTINGS_QUERY })
+  if (!data && error) throw error
+
+  const sentry = Sentry.use()
+  React.useEffect(() => {
```

Comment:
If an error is thrown, doesn't React complain that it's a different number of hooks between renders?

Reply:
In other words, it seems that `useEffect` should be after `if (condition) throw`

Reply:
> If an error is thrown, doesn't React complain that it's a different number of hooks between renders?

it would, but it throws and exception and crashes anyways, and this case is more like an assertion -- this should never happen and if it happens it indicates an application error and should be handled by developers, it's not an expected behaviour

> in other words, it seems that useEffect should be after if (condition) throw

but it is